src/middleware/uploadHandler.js

```diff
@@ -0,0 +1,58 @@
+const multer = require('multer');
+const fs = require('mkdir-recursive');
+const helper = require('../utils/helper');
+const multerS3 = require('multer-s3');
+const { s3 } = require('../utils/awsSDK');
+
+const isvalidate = (folder) => {
+  fs.mkdirSync(folder, true);
+
+  return folder;
+};
+
+const storage = multer.diskStorage({
+  destination(req, file, cb) {
+    const folder = helper.getNewfolder('my-uploads', 'local');
+    cb(null, isvalidate(folder));
+  },
+  filename(req, file, cb) {
+    cb(null, `${file.originalname}-${Date.now()}`);
+  },
+});
+
+const aws = multer({
+  storage: multerS3({
+    s3,
+    bucket: helper.getNewfolder('my-uploads', 'aws'),
+    acl: 'public-read',
+    metadata: (request, file, cb) => {
+      cb(null, { fieldName: file.fieldname });
+    },
+    key: (request, file, cb) => {
+      cb(null, 'uploadedFile');
+    },
+  }),
+});
+
+const uploadHandlerLocal = multer({ storage });
+
+const uploadHandlerAws = multer({ aws });
+
+const uploadHandler = type => (req, res, next) => {
```

Comment:
you could call this uploadMiddleware, as you used req, res, next you basically created a middleware, right?